syntax = "proto3";
package csce438;

import "google/protobuf/timestamp.proto";
import "coordinator.proto";
// ------------------------------------------------------------
// The coordinator service definition
// These are the different functionalities exposed by service
// ------------------------------------------------------------

//Init and Heartbeat potentially redundant
service SynchService{
//   rpc Heartbeat (ServerInfo) returns (Confirmation) {}
//   rpc GetServer (ID) returns (ServerInfo) {}
  rpc SendFollowingList(UserInfo) returns (Confirmation) {}
  rpc SendTimeline(TimeInfo) returns (Confirmation) {}
  rpc ResynchServer(ServerInfo) returns (Confirmation) {}
  // ZooKeeper API here
}

//server info message definition
// message ServerInfo{
//   int32 serverID = 1;
//   string hostname = 2;
//   string port = 3;
//   string type = 4;
//   int32 cluster = 5;
// }

// //confirmation message definition
// message Confirmation{
//   bool status = 1;
// }

// //id message definition
// message ID{
//   int32 id = 1;
// }

message UserInfo{
    string following = 1;
}

message TimeInfo{
  string username = 1;
  string timeline = 2;
}


